browser/page_mapping.go

```diff
@@ -206,18 +207,57 @@ func mapPage(vu moduleVU, p *common.Page) mapping { //nolint:gocognit,cyclop
 		"on": func(event string, handler sobek.Callable) error {
 			tq := vu.taskQueueRegistry.get(vu.Context(), p.TargetID())
 
-			mapMsgAndHandleEvent := func(m *common.ConsoleMessage) error {
-				mapping := mapConsoleMessage(vu, m)
-				_, err := handler(sobek.Undefined(), vu.Runtime().ToValue(mapping))
-				return err
-			}
-			runInTaskQueue := func(m *common.ConsoleMessage) {
-				tq.Queue(func() error {
-					if err := mapMsgAndHandleEvent(m); err != nil {
-						return fmt.Errorf("executing page.on handler: %w", err)
+			var runInTaskQueue func(any)
+			switch event {
+			case common.EventPageConsoleAPICalled:
+				mapMsgAndHandleEvent := func(m *common.ConsoleMessage) error {
+					mapping := mapConsoleMessage(vu, m)
+					_, err := handler(sobek.Undefined(), vu.VU.Runtime().ToValue(mapping))
+					return err
+				}
+				runInTaskQueue = func(a any) {
+					tq.Queue(func() error {
+						m, ok := a.(*common.ConsoleMessage)
```

Comment:
I've linked this comment to a [comment](https://github.com/grafana/xk6-browser/issues/1227#issuecomment-2393348036) in the generalize `page.on` events [issue](https://github.com/grafana/xk6-browser/issues/1227).

We can tackle the generalization after merging in this PR 👍 

I'll resolve this comment so that I can work on the other feedback.